src/i965_encoder.c

```diff
@@ -808,8 +810,28 @@ intel_encoder_check_misc_parameter(VADriverContextP ctx,
         VAEncMiscParameterBufferQualityLevel* param_quality_level = (VAEncMiscParameterBufferQualityLevel*)pMiscParam->data;
         encoder_context->quality_level = param_quality_level->quality_level;
 
-        if (encoder_context->quality_level == 0)
-            encoder_context->quality_level = ENCODER_DEFAULT_QUALITY;
+        if (encoder_context->quality_level == 0) {
```

Comment:
no, this quality_level is from extern APIs and can change the value set by intel_enc_hw_context_init()  . And value 0 means driver should choose its own quality level automatically